crates/miden-lib/src/note/well_known_note.rs

```diff
@@ -157,4 +161,145 @@ impl WellKnownNote {
             },
         }
     }
+
+    /// Checks the correctness of the provided note inputs against the target account.
+    ///
+    /// It returns the corresponding note consumption status in case we can guarantee that the note
+    /// cannot be consumed, or `None` otherwise.
+    ///
+    /// It performs:
+    /// - for all notes: a
```

Comment:
This looks incomplete. Perhaps we could move this here:
> check that note inputs have the correct number of values.

Reply:
I was hesitating about using the `for all notes` wording here, because strictly speaking we don't perform any checks for the `SWAP` note, so I decided to use a duplicating lines (but forgot to remove this `for all notes:` variant). 

Maybe we can say `for P2ID and P2IDE notes` instead of `for all notes`? That will be a bot more clumsy, but formally correct.

Reply:
Ah, good call, in that case leaving it separate for each note does seem cleaner actually.